applications/ConvectionDiffusionApplication/custom_elements/laplacian_embedded_element.h

```diff
@@ -0,0 +1,390 @@
+        MatrixType& rLeftHandSideMatrix,
+        VectorType& rRightHandSideVector,
+        const ProcessInfo& rCurrentProcessInfo,
+        const EmbeddedElementData& rData);
+
+    ///@}
+    ///@name Protected  Access
+    ///@{
+
+
+    ///@}
+    ///@name Protected Inquiry
+    ///@{
+
+
+    ///@}
+    ///@name Protected LifeCycle
+    ///@{
+
+    // Protected default constructor necessary for serialization
+    LaplacianEmbeddedElement() : LaplacianElement()
+    {
+    }
+
+    ///@}
+
+private:
+    ///@name Static Member Variables
+    ///@{
+
+    ///@}
+    ///@name Member Variables
+    ///@{
+
+
+    ///@}
+    ///@name Serialization
+    ///@{
+
+    friend class Serializer;
+
+    void save(Serializer& rSerializer) const override
+    {
+        KRATOS_SERIALIZE_SAVE_BASE_CLASS(rSerializer, LaplacianElement);
+    }
+
+    void load(Serializer& rSerializer) override
+    {
+        KRATOS_SERIALIZE_LOAD_BASE_CLASS(rSerializer, LaplacianElement);
+    }
+
+    ///@}
+    ///@name Private Operators
+    ///@{
+
+
+    ///@}
+    ///@name Private Operations
+    ///@{
+
+
+    ///@}
+    ///@name Private  Access
+    ///@{
+
+
+    ///@}
+    ///@name Private Inquiry
+    ///@{
+
+
+    ///@}
+    ///@name Un accessible methods
+    ///@{
+
+    /// Assignment operator.
+    //LaplacianEmbeddedElement& operator=(const LaplacianEmbeddedElement& rOther);
+
+    /// Copy constructor.
+    //LaplacianEmbeddedElement(const LaplacianEmbeddedElement& rOther);
+
+    ///@}
+
+}; // Class LaplacianEmbeddedElement
+
+namespace LaplacianEmbeddedInternals {
+
+template <size_t TDim, size_t TNumNodes>
+ModifiedShapeFunctions::Pointer GetContinuousShapeFunctionCalculator(
+    const Element &rElement,
+    const Vector &rNodalDistances);
+
+template<std::size_t TDim>
+class EmbeddedElementData
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    static constexpr std::size_t NumNodes = TDim + 1;
+
+    typedef GeometryData::ShapeFunctionsGradientsType ShapeFunctionsGradientsType;
+    typedef std::vector<array_1d<double,3>> InterfaceNormalsType;
+    typedef array_1d<double,NumNodes> NodalScalarData;
+
+    ///@}
+    ///@name Public Members
+    ///@{
+
+    double PenaltyCoefficient;
+
+    NodalScalarData NodalDistances;
+
+    Matrix PositiveSideN;
+    ShapeFunctionsGradientsType PositiveSideDNDX;
+    Vector PositiveSideWeights;
+
+    Matrix PositiveInterfaceN;
+    ShapeFunctionsGradientsType PositiveInterfaceDNDX;
+    Vector PositiveInterfaceWeights;
+    InterfaceNormalsType PositiveInterfaceUnitNormals;
+
+    std::vector< size_t > PositiveIndices;
+
+    size_t NumPositiveNodes;
+    size_t NumNegativeNodes;
+
+    ///@}
+    ///@name Public Operations
+    ///@{
+
+    /**
+     * @brief Split element data container initialization
+     * This method initializes the embedded formulation data container. This implies to get the nodal distances.
+     */
+    void Initialize(
+        const Element& rElement
+    );
```

Comment:
```suggestion
    void Initialize(const Element& rElement);
```
One line for single argument functions (style guide).